crates/autopilot/src/arguments.rs

```diff
@@ -340,32 +341,28 @@ impl std::fmt::Display for Arguments {
     }
 }
 
-#[derive(clap::Parser, Debug, Clone)]
-pub struct FeePolicy {
-    /// Type of fee policy to use. Examples:
-    ///
-    /// - Surplus without cap
-    /// surplus:0.5:1.0
-    ///
-    /// - Surplus with cap:
-    /// surplus:0.5:0.06
-    ///
-    /// - Price improvement without cap:
-    /// price_improvement:0.5:1.0
-    ///
-    /// - Price improvement with cap:
-    /// price_improvement:0.5:0.06
-    ///
-    /// - Volume based:
-    /// volume:0.1
-    #[clap(long, env, default_value = "surplus:0.0:0.9")]
-    pub fee_policy_kind: FeePolicyKind,
-
-    /// Should protocol fees be collected or skipped for orders whose
-    /// limit price at order creation time suggests they can be immediately
-    /// filled.
-    #[clap(long, env, action = clap::ArgAction::Set, default_value = "true")]
-    pub fee_policy_skip_market_orders: bool,
+/// A fee policy to be used for orders base on it's class.
+/// Examples:
+/// - Surplus with a high enough cap for limit orders
+/// surplus:0.5:0.9:limit
+///
+/// - Surplus with cap for market orders:
+/// surplus:0.5:0.06:market
+///
+/// - Price improvement with a high enough cap for market orders:
+/// price_improvement:0.5:0.9:market
```

Comment:
I find this  parsing a bit hard to read. Would slightly prefer separate params.

`--in-market-order-fees surplus:0.5:0.06 --out-of-market-order-fees price_improvement:0.5:0.9`

Reply:
That was the initial approach, then it was changed to a unified version in order to support multiple fee policies: https://github.com/cowprotocol/services/pull/2550#discussion_r1533761788

Reply:
In this case, the "fee policy order class" would have to be optional imo (ie allow `price_improvement:0.5:0.9` which would take effect for all orders). 

Otherwise I don't see what we benefit. In the old model we could also specify multiple fee policies (by using a `Vec` for both configs).

Reply:
Added `any` type.